### ISSUE TITLE
Use  GCNConv to make MNIST classification

### ISSUE THREAD

**Issue by tonyandsunny** [rusty1s/pytorch_geometric#747]
## ❓ Questions & Help

Hi, thank you very much for this framework. I just changed SplineConv to GCNConv in mnist_graculs.py. But the model loss does not fall, loss does not converge. I'm confused.
Could you give me some advices?

**Comment by rusty1s** [comment]
The `GCNConv` is not really designed for this task, so a poor performance is to be expected since the model does not take any spatial information into account and neighborhood connections does not provide any information without it. I can confirm that simply replacing the conv operators does not work (I trained the network until epoch 4 and it reaches a loss of 2 with test accuracy of 23%). 

You can increase its performance slightly by passing the node positions as input features though.

**Comment by tonyandsunny** [comment]
Thank you very much for your reply so quickly.
"The node positions as input features"-Can we use the connective nodes positions e
(−
||pi −p j || 2
)
2σ 2
as the weight
There are GCN and Monet in MNIST and cifar-10 dataset in "Image Classification with
Hierarchical Multigraph Networks". Therefore, GCN should be used to solve image classification tasks. Could you give me some suggestions?

**Comment by tonyandsunny** [comment]
Hi, have you used ArmaConv to do MNIST classification tasks?
I used the Gaussian kernel of the difference of pixel coordinates as the weight of the edge as described in the paper, but this result is not good, I am very confused.

**Comment by rusty1s** [comment]
Yes, you can also use edge weights to encode distances for `GCNConv` or `ChebConv`. That is exactly what your linked paper is doing (in addition to using node coordinates as input features). I think 85% should be easily doable with `GCNConv` when following the experimental setup of the paper.

**Comment by tonyandsunny** [comment]
def normalized_cut_2d_adj(edge_index, pos):
    row, col = edge_index
    weight = torch.norm(pos[row] - pos[col], p=2, dim=1)
    weight = torch.exp(-weight/2*0.01)
    return weight
class Net(nn.Module):
    def __init__(self, nfeat, nhid, nclass, dropout):
        super(Net, self).__init__()
        self.conv1 = ARMAConv(
            d.num_features,
            32,
            num_stacks=3,
            num_layers=2,
            shared_weights=True,
            dropout=0.25)
        self.conv2 = ARMAConv(
            32,
            64,
            num_stacks=3,
            num_layers=2,
            shared_weights=True,
            dropout=0.25,
            act=None)
        # self.conv1 = SplineConv(d.num_features, 32, dim=2, kernel_size=5)
        # self.conv2 = SplineConv(32, 64, dim=2, kernel_size=5)
        self.dropout = dropout
        self.fc1 = torch.nn.Linear(64, 512)
        self.fc2 = torch.nn.Linear(512, d.num_classes)

    def forward(self, x, adj,batch, data):
        # plt.imshow(adj.to_dense().cpu().numpy()[:75,:75])
        # plt.colorbar()
        # plt.show()
        weight1 = normalized_cut_2d_adj(data.edge_index, data.pos)
        data.x = F.elu(self.conv1(data.x, data.edge_index,weight1))

        #weight = normalized_cut_2d(data.edge_index, data.pos)
       
        #cluster = graclus(data.edge_index, weight, data.x.size(0))
        cluster = graclus(data.edge_index)
        data.edge_attr = None
        data = max_pool(cluster, data, transform=T.Cartesian(cat=False))
        
        weight2 = normalized_cut_2d_adj(data.edge_index, data.pos)
        data.x = F.elu(self.conv2(data.x , data.edge_index,weight2))

        #weight = normalized_cut_2d(data.edge_index, data.pos)
        # cluster = voxel_grid(data.pos, data.batch, size=7, start=0, end=28)
        #cluster = graclus(data.edge_index, weight, data.x.size(0))
        cluster = graclus(data.edge_index)
        x, batch = max_pool_x(cluster, data.x, data.batch)
        x = global_mean_pool(x, batch)
        x = F.elu(self.fc1(x))
        x = F.dropout(x, self.dropout, training=self.training)
        x = self.fc2(x)
        return F.log_softmax(x, dim=1)

This is my main part of the code, but the loss doesn't seem to be converge, and the accuracy doesn't improve.

**Comment by tonyandsunny** [comment]
@rusty1s 
I am trying again the method you said today, the loss convergence and accuracy is improved, thank you very much. 
Maybe you can close this question.

**Comment by rusty1s** [comment]
Awesome :)
